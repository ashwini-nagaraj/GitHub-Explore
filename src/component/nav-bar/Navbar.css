.avatar {
  border-radius: 50%;
  object-fit: cover;
}

.navbar {
  margin: 1px;
  background-color: rgb(54, 50, 50);
}

.search-box-user {
  border-radius: 5px;
  height: 39px;
  width: 250px;
  background: white;
  font-size: 15px;
  outline: none !important;
  float: right;
}

@media screen and (max-width: 780px) {
  .navbar-nav {
    margin: 1px auto !important;
  }
  .nav-item {
    margin: 1px auto !important;
  }
}
