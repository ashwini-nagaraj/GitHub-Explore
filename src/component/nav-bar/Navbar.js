import React, { useState } from "react";
import "./Navbar.css";

const logoImg =
  "https://github.githubassets.com/images/modules/logos_page/Octocat.png";

function Navbar({ user }) {
  const [inputText, setInputText] = useState("");
  const startSearch = (userName) => {
    window.location.href = `/${userName}`;
  };
  return (
    <nav data-testid="navbar" className="navbar navbar-expand-sm .bg-dark.bg-gradient">
      <ul data-testid="list-ul" class="navbar-nav ml-auto">
        <li data-testid="li" class="nav-item">
          <img
            data-testid="profileImg"
            src={logoImg}
            width="40"
            height="40"
            className="mr-3 avatar"
            alt="profile"
          />
        </li>
        <li data-testid="list-items" class="nav-item px-3">
          <input
            data-testid="search box"
            className="search-box-user"
            type="text"
            placeholder="Search for other user"
            onChange={(e) => {
              setInputText(e.target.value);
            }}
            onKeyDown={(e) => {
              if (e.keyCode === 13) {
                e.preventDefault();
                startSearch(inputText);
              }
            }}
          ></input>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
