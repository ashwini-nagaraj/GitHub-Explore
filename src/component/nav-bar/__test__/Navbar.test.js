import React from "react";
import ReactDOM from "react-dom";
import Navbar from "../Navbar";
import { render , cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashig", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Navbar></Navbar>, div);
});

it("renders loading correctly", () => {
  const { getByTestId } = render(
    <Navbar>
    </Navbar>
  );
  const navbar = getByTestId("navbar");
  const listUl = getByTestId("list-ul");
  expect(navbar).toContainElement(listUl);
});


it("matches snapshot",()=>{
   const tree=  renderer.create(<Navbar></Navbar>).toJSON();
    expect(tree).toMatchSnapshot();
})