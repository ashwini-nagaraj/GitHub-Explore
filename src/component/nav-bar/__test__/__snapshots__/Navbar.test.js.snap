// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm .bg-dark.bg-gradient"
  data-testid="navbar"
>
  <ul
    class="navbar-nav ml-auto"
    data-testid="list-ul"
  >
    <li
      class="nav-item"
      data-testid="li"
    >
      <img
        alt="profile"
        className="mr-3 avatar"
        data-testid="profileImg"
        height="40"
        src="https://github.githubassets.com/images/modules/logos_page/Octocat.png"
        width="40"
      />
    </li>
    <li
      class="nav-item px-3"
      data-testid="list-items"
    >
      <input
        className="search-box-user"
        data-testid="search box"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search for other user"
        type="text"
      />
    </li>
  </ul>
</nav>
`;
